docs/onboarding.md

```diff
@@ -0,0 +1,157 @@
+
+The CLI also allows to specify which source to use for `onboarding` and `home` applications. The defaults will be hosted on `github.com/cozy`.
+
+After creation, an instance has a `registerToken` generated randomly.
+
+All cozy-instance fields are considered private unless explictly prefixed by `public-`. Public instance field can be seen from the `instance` endpoint. Only public fields we can think of for now are `public-name`, `public-locale`.
+
+## Onboarding steps
+
+This document and the cozy-stack are only concerned with login and password registering step which are important for security.
+
+All other steps are handled by the `onboarding` application.
+
+The `onboarding` application SHOULD therefore provide the following features
+- When started with a `registerToken`, allow the user to create a password
+- When started with a `contextToken` ([see auth doc](./auth.md#how-to-get-a-token)) use it to retrieve instance document.
+  - If the instance document is complete **according to the `onboarding` app**, redirect to `home` application.
+  - Otherwise, performs whatever steps it deems necessary to fill out the instance (ask for user email, help set up `myaccounts` accounts, say thank you...)
+
+This makes cozy-stack simple and safer while allowing behaviour modification for several install types by picking the correct `onboarding` application / branch.
+
+This makes it easier to add more onboarding steps and have them run on already-installed cozy : On next login after onboarding application update, it will ask the user.
+
+## Redirections
+
+When an user attempts to access the root of its instance (`https://example.cozycloud.cc`) or an application (`https://contacts.example.cozycloud.cc`), and she is not logged-in, she is redirected :
+
+- If the instance has a `password` set, to the `/login` page
+- If the instance has a `registerToken` set, to the `onboarding` application.
+
+After login, the user is always redirected to the `onboarding` application. It is the `onboarding` application responsibility to check if registering is complete and reredirect to home.
+
+
+## Necessary new routes
+
+### GET /instance
+
+If the user is not logged in, display public instance information
+#### Request
+```http
+GET https://alice.example.com/instance
+```
+#### Response
+```json
+{
+  "public-locale":"fr",
+  "public-name":"Alice Martin"
+}
+```
+
+If the user is logged in, display all instance information, except password
+#### Request
+```
+GET https://alice.example.com/instance
+Cookie sessionid=xxxx
+```
+#### Response
+```json
+{
+  "_rev":"3-56521545485448482",
+  "domain": "alice.example.com",
+  "owner-email": "alice@example.com",
+  "public-locale":"fr",
+  "public-name":"Alice Martin"
+}
+```
+
+### POST /instance
+
+If the user is not logged in, allow to set the password
```

Comment:
Maybe here, if I understand correctly, if the user is not logged in, that should allow to set the password if and only if the registerToken used by the user is correct, isn't it?